src/access.rs

```diff
@@ -0,0 +1,438 @@
+				}
+
+				let block_status = client.get_block_status(&block_hash).await?;
+				if let Some(block_height) = block_status.height {
+					return Ok(Some(ConfirmedTx { tx, block_header, pos, block_height }));
+				} else {
+					// If any previously-confirmed block suddenly is no longer confirmed, we found
+					// an inconsisteny and should start over.
+					return Err(Error::ChainAccessInconsistency);
+				}
+			}
+		}
+		Ok(None)
+	}
+
+	async fn sync_unconfirmed_transactions(
+		&self, confirmables: &Vec<&(dyn Confirm + Sync)>,
+	) -> Result<(), Error> {
+		let client = &*self.blockchain;
+		// Query the interface for relevant txids and check whether the relevant blocks are still
+		// in the best chain, mark them unconfirmed otherwise. If the transactions have been
+		// reconfirmed in another block, we'll confirm them in the next sync iteration.
+		let relevant_txids = confirmables
+			.iter()
+			.flat_map(|c| c.get_relevant_txids())
+			.collect::<HashSet<(Txid, Option<BlockHash>)>>();
+		for (txid, block_hash_opt) in relevant_txids {
+			if let Some(block_hash) = block_hash_opt {
+				let block_status = client.get_block_status(&block_hash).await?;
+				if block_status.in_best_chain {
+					// Skip if the block in question is still confirmed.
+					continue;
+				}
+			}
+
+			for c in confirmables {
+				c.transaction_unconfirmed(&txid);
+			}
+		}
+
+		Ok(())
+	}
+
+	pub(crate) fn create_funding_transaction(
+		&self, output_script: &Script, value_sats: u64, confirmation_target: ConfirmationTarget,
+	) -> Result<Transaction, Error> {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fee_rate = self.blockchain.estimate_fee(num_blocks)?;
+
+		let locked_wallet = self.wallet.lock().unwrap();
+		let mut tx_builder = locked_wallet.build_tx();
+
+		tx_builder.add_recipient(output_script.clone(), value_sats).fee_rate(fee_rate).enable_rbf();
+
+		let (mut psbt, _) = tx_builder.finish()?;
+		log_trace!(self.logger, "Created funding PSBT: {:?}", psbt);
+
+		// We double-check that no inputs try to spend non-witness outputs. As we use a SegWit
+		// wallet descriptor this technically shouldn't ever happen, but better safe than sorry.
+		for input in &psbt.inputs {
+			if input.witness_utxo.is_none() {
+				log_error!(self.logger, "Tried to spend a non-witness funding output. This must not ever happen. Panicking!");
+				panic!("Tried to spend a non-witness funding output. This must not ever happen.");
+			}
+		}
+
+		if !locked_wallet.sign(&mut psbt, SignOptions::default())? {
+			return Err(Error::FundingTxCreationFailed);
+		}
+
+		Ok(psbt.extract_tx())
+	}
+
+	pub(crate) fn get_new_address(&self) -> Result<bitcoin::Address, Error> {
+		let address_info = self.wallet.lock().unwrap().get_address(AddressIndex::New)?;
+		Ok(address_info.address)
+	}
+}
+
+struct ConfirmedTx {
+	tx: Transaction,
+	block_header: BlockHeader,
+	block_height: u32,
+	pos: usize,
+}
+
+impl<D> FeeEstimator for ChainAccess<D>
```

Comment:
think this is supposed to be `LdkLiteChainAccess`

Reply:
Ah, good catch but actually not. I decided to drop the `LdkLite` prefix from all structs in #11, and to minimize rebasing efforts I kinda snug this into here, too (as nothing depends on it).